rdrf/rdrf/filestorage.py

```diff
@@ -1,39 +1,76 @@
-# Attempt to clean up the interaction with GridFS
 import logging
+import re
+import gridfs
+from bson.objectid import ObjectId
+from .models import Registry, CDEFile
```

Comment:
CDEFile model has link to section but what  if the linked section is a "multisection?

Do we need a nullable index field also ( ie null in the case where the section is a "normal" (non-)multisection 0-numitems-1 otherwise 

After looking I can see there's no need as it's deducible from the mongo record ) ( as is the fact that it "belongs" to a given patient  - We can report on mongo cdes including file cdes using the explorer so this shouldn't matter ( although some queries would be easier eg find all patients who haven't uploaded a catscan file ))